### PR TITLE
adding Service Spec editor for admin/settings

### PR DESCRIPTION
**Pull request by jeremyd**

### CODE REVIEW

File: dashboard/src/actions/index.js [hunk 1]
```diff
@@ -723,3 +723,81 @@ export const fetchServiceSpecs = (project_slug) => {
...
+export const SERVICE_SPECS_SETTINGS_FETCH_REQUEST = 'SERVICE_SPECS_SETTINGS_FETCH_REQUEST'
+export const SERVICE_SPECS_SETTINGS_FETCH_SUCCESS = 'SERVICE_SPECS_SETTINGS_FETCH_SUCCESS'
+export const SERVICE_SPECS_SETTINGS_FETCH_FAILURE = 'SERVICE_SPECS_SETTINGS_FETCH_FAILURE'
+
+export const fetchServiceSpecSettings = () => {
+  return {
+    [CALL_API]: {
+      endpoint: `${API_ROOT}/admin/serviceSpecs`,
+      method: 'GET',
+      headers: {
```
Comment (geoffrey) [inline dashboard/src/actions/index.js]: nit: seems like headers are the same for all API calls here we could extract in a function to DRY it up

File: dashboard/src/components/admin/ServiceSpecForm.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+const ServiceSpecForm = reduxForm({
+  form: 'serviceSpec',
+  validate
+})(ServiceSpec)
+
+export default connect( state => {
+  const formValues = state.form.serviceSpec
+  return { formValues: formValues }
```
Comment (geoffrey) [inline dashboard/src/components/admin/ServiceSpecForm.js]: nit `return { formValues }`

File: dashboard/src/components/admin/ServiceSpecList.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    }
+    this.renderSpecs = this.renderSpecs.bind(this)
+    this.onSaveSpec = this.onSaveSpec.bind(this)
+    this.onAddSpec = this.onAddSpec.bind(this)
+    this.onDeleteSpec = this.onDeleteSpec.bind(this)
+  }
+
+  componentWillMount() {
+    this.props.fetchServiceSpecSettings()
+
```
Comment (geoffrey) [inline dashboard/src/components/admin/ServiceSpecList.js]: nit: extra line

File: dashboard/src/components/admin/ServiceSpecList.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      </li>
+    )
+  }
+
+  renderEditSpec(specSetting) {
+    return(
+      <li className="list-group-item" key={specSetting._id}>
+        <div className="feed-element">
+          <div className="media-body">
+            <ServiceSpecForm initialValues={specSetting} edit={specSetting.edit} onSubmit={this.onSaveSpec} onCancel={ () => this.onCancel()} onDelete={this.onDeleteSpec}/>
```
Comment (geoffrey) [inline dashboard/src/components/admin/ServiceSpecList.js]: nit: any reason you don't have `onCancel={ thi.onCancel }` ?

Comment (jeremyd) [inline dashboard/src/components/admin/ServiceSpecList.js]: cause it needs to bind to this.  the alternative might be to bind it in the constructor like some of the onSave type stuff

File: dashboard/src/actions/index.js (hunk 1 above)
Comment (jeremyd) [inline dashboard/src/actions/index.js]: I tried to do this, but failed...  Not sure how to pass this around in this special formatting here...

**Pull request merged**